lms/product/plugin/grading.py

```diff
@@ -0,0 +1,12 @@
+class GradingPlugin:
+    @staticmethod
+    def configure_grading_for_launch(request, js_config, assignment):
+        """Configure grading during a launch."""
+        if request.lti_user.is_instructor:
+            # For instructors, display the toolbar
+            js_config.enable_instructor_toolbar(enable_grading=assignment.is_gradable)
+        else:
+            # Create or update a record of LIS result data for a student launch
+            # We'll query these rows to populate the student dropdown in the
+            # instructor toolbar
+            request.find_service(name="grading_info").upsert_from_request(request)
```

Comment:
This method seems to do quite a few different possible things:

 * Show the toolbar with grading support
 * Show the toolbar for editing the assignment **<<< Nothing to do with grading at all?**
 * Store grading info for the user

So as far as the name goes: "configure_**grading**\_for_**launch**":

 * Grading applies 2/3 cases
 * Launch applies 3/3 cases

This makes me wonder if this is more of a launch plugin than a grading plugin? It's also only called from the launch view.

Some different potential ways of skinning this cat:

* A `LaunchPlugin` with `show_instructor_toolbar` and `configure_grading_for_student`
   * Canvas would do nothing for `show_instructor_toolbar` and put everything in `configure_grading_for_student`
   * We could also move the test for `is_instructor` outside of the plugins
* A `LaunchPlugin` with a more generic "pre-launch-hook" point

I think I prefer the more structured one though, and I think the commonality of checking for the student or not is correct to keep outside (at least as far as the examples we have so far.

I think if we had a `LaunchPlugin` that some of the other misc items could move there too.

Reply:
Split the toolbar options in two, editing and grading. Quite like the explicitly of this now :+1: